client/src/unifyfs.c

```diff
@@ -698,60 +704,54 @@ static int process_read_data(read_req_t* read_reqs, int count, int* done)
             }
 
             /* this reply overlaps with the request, check that
-             * we didn't get an error */
-            if (rep->errcode != UNIFYFS_SUCCESS) {
```

Comment:
This check on setting the read request errcode field is important, because a single read request could have two replies.  If the first reply has an error, but the second one succeeded, we still need to be sure the read request ends up marked with an error.  Basically, if *any* reply has an error for the read request, the read request should have an error.

Reply:
@adammoody I kept this code in the new commit.